src/ui/public/visualize/loader/embedded_visualize_handler.ts

```diff
@@ -126,6 +129,8 @@ export class EmbeddedVisualizeHandler {
     }
     this.uiState = this.vis.getUiState();
 
+    this.handlers = { uiState: this.uiState, onDestroy: fn => (this.handlers.destroyFn = fn) };
```

Comment:
handlers needs to be passed to renderers and should have `onDestroy` and `done` methods (todo)